test/distributed/checkpoint/test_state_dict.py

```diff
@@ -443,6 +448,21 @@ def is_cpu(v):
             self.assertEqual(mst, {})
             self.assertEqual(ost, {})
 
+    @with_comms
+    @skip_if_lt_x_gpu(1)
+    def test_activation_ckpt_fqns(self) -> None:
+        """Tests that activation checkpointing prefixes are removed from module names"""
+        model = CompositeParamModel(device=torch.device("cuda"))
+        apply_activation_checkpointing(model)
+        model_names = [(model, name) for name, _ in model.named_parameters()]
+
+        ddp_model = DDP(model)
+        ddp_names = [(ddp_model, name) for name, _ in ddp_model.named_parameters()]
+
+        for model, name in model_names:
+            for fqn in _get_fqns(model, name):
+                self.assertNotIn(_CHECKPOINT_WRAPPED_MODULE, fqn)
```

Comment:
We should just call `get_state_dict()` and compare its keys with the original model without activation checkpoint (you can deepcopy the original model).